libs/barista-components/table/src/table.ts

```diff
@@ -212,8 +217,11 @@ export class DtTable<T> extends _DtTableBase<T> implements OnDestroy {
   @ViewChild(CdkPortalOutlet, { static: true })
   _portalOutlet: CdkPortalOutlet;
 
-  /** @internal A list of columns */
-  @ContentChildren(DtSimpleColumnBase) _childColumns: QueryList<
+  /** @internal A list of complex columns */
+  @ContentChildren(DtHeaderCell, { descendants: true })
+  _childComplexColumns: QueryList<DtHeaderCell>;
```

Comment:
The presence of a HeaderCell makes it a complex definition? What you mean by this is that you are querying for non simple columns, is  this correct?

Reply:
Correct. Would you prefer something like `nonSimple` instead of complex?

Reply:
to be quite honest, I don't know what I would prefer here. At least a comment that outlines why this is necessary would be benefitial.